Dfe.SchoolAccount.Web.Tests/Authorization/FailedAuthorizationMiddlewareResultHandlerTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿namespace Dfe.SchoolAccount.Web.Tests.Authorization;
+
+using Dfe.SchoolAccount.Web.Authorization;
+using Dfe.SchoolAccount.Web.Constants;
+using Dfe.SchoolAccount.Web.Tests.Helpers;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authorization.Policy;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+
+[TestClass]
+public sealed class FailedAuthorizationMiddlewareResultHandlerTests
+{
+    #region Task HandleAsync(RequestDelegate, HttpContext, AuthorizationPolicy, PolicyAuthorizationResult)
+
+    [TestMethod]
+    public async Task HandleAsync__RedirectsToYourInstitutionIsNotYetEligibleForThisServicePage__WhenFailureOccurs()
+    {
+        var logger = new NullLogger<FailedAuthorizationMiddlewareResultHandler>();
+        var defaultHandlerMock = new Mock<IAuthorizationMiddlewareResultHandler>();
+        var failedAuthorizationMiddlewareResultHandler = new FailedAuthorizationMiddlewareResultHandler(logger, defaultHandlerMock.Object);
+
+        var next = (RequestDelegate)(context => Task.CompletedTask);
+
+        var responseMock = new Mock<HttpResponse>();
+        var contextMock = new Mock<HttpContext>();
+        contextMock.SetupGet(mock => mock.Response)
+            .Returns(responseMock.Object);
+
+        var policy = (AuthorizationPolicy)null!;
+        var authorizeResult = PolicyAuthorizationResult.Forbid(
+            AuthorizationFailure.Failed(new AuthorizationFailureReason[] {
+                new AuthorizationFailureReason(null!, AuthorizationFailureConstants.YourInstitutionIsNotYetEligibleForThisService),
+            })
+        );
+
+        await failedAuthorizationMiddlewareResultHandler.HandleAsync(next, contextMock.Object, policy, authorizeResult);
+
+        responseMock.Verify(mock => mock.Redirect(It.Is<string>(param => "/your-institution-is-not-yet-eligible-for-this-service" == param)));
```

Comment:
Can this not reference the constant `ErrorPageConstants.YourInstitutionIsNotYetEligibleForThisServiceHandle` instead?

Reply:
I think it is useful since it acts as a test for the URL construction.